internal/ir/xds.go

```diff
@@ -1651,3 +1652,38 @@ type HTTPTimeout struct {
 	// The maximum duration of an HTTP connection.
 	MaxConnectionDuration *metav1.Duration `json:"maxConnectionDuration,omitempty" yaml:"maxConnectionDuration,omitempty"`
 }
+
+// TLSBundle contains tls certificate, private key and ca file in []byte format.
+// +k8s:deepcopy-gen=true
+type TLSBundle struct {
+	Hostname string
+	CACert   TLSCACertificate
+	TLSCert  *TLSCertificate
+}
+
+//func (b TLSBundle) GetXdsCertSecretName() string {
+//	return NameWithSuffix(b.TLSCertificate.Name, "tls")
+//}
+
+//func (b TLSBundle) GetXdsCaSecretName() string {
+//	return NameWithSuffix(b.TLSCACertificate.Name, "ca")
+//}
+
+func (b TLSBundle) Mtls() bool {
+	if b.TLSCert != nil && b.TLSCert.ServerCertificate != nil && b.TLSCert.PrivateKey != nil {
```

Comment:
Should we support MTLS in the scope of the current PR?

Reply:
no we shouldnt, should be fixed with 
https://github.com/envoyproxy/gateway/issues/2536

Reply:
so should i remove everything related to serverCert and private key ? @arkodg

Reply:
yah, its not part of the spec yet

Reply:
removed